plugins/ua_securitypolicy_basic128rsa15.h

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef UA_SECURITYPOLICY_BASIC128RSA15_H_
+#define UA_SECURITYPOLICY_BASIC128RSA15_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ua_plugin_securitypolicy.h"
+#include "ua_plugin_log.h"
+
+typedef int(*UA_EntropySourceFunc_sp_basic128rsa15)(void *data, unsigned char *output,
+                                                    size_t len, size_t *olen);
+
+/** TODO: document
+ * For more information see mbedtls documentation on mbedtls entropy
+ */
+typedef struct {

```

Comment:
These seem like private definitions. Put on top of the .c file.
And leave just a skeleton API for initialization..

Reply:
I don't quite get what you mean. Could you elaborate?

Reply:
Users of the plugin don't need to know the structures and the method signature.
So hide it from them.
The definitione can live in the .c file. No need to keep them in the public header.

Reply:
This was intended to be supplied by the user. The user has the possibility of supplying entropy sources and functions to mbedtls. Since this differs from platform to platform it cannot be implemented in a plugin. The user can also choose to not supply any, in which case he just passes NULL and no entropy functions are initialized to mbedtls.

Reply:
Simplify, simplify, simplify.

I like the idea to empower users add entropy sources. I just fear the we open a huge can of worms where users have doubts about how secure the default settings are. I think we can cover 95% of use cases with our default implementation. And the remaining users are on special platforms where they fork+modify the plugins anyway.

I found the following in the mbedTLS docs: https://tls.mbed.org/kb/how-to/how-do-i-port-mbed-tls-to-a-new-environment-OS

- On Windows and different Unix platforms that provide /dev/urandom, a default OS-based source is registered. It can be disabled at compile-time without affecting the rest of the library.
- For obvious security reasons, the entropy module will refuse to output anything until a declared-strong source has been registered.

That sounds like we can have a fool-proof API.
- Use the default strong entropy sources on Windows/Unix-like.
- On all other platforms emit a compiler #warning + refuse to create entropy + state the missing entropy source in log output.
- Document the place where entropy sources shall be added inside the plugin, not in the plugin header. Then we can use the definitions from mbedtls and don't have to wrap everything.

This is a sensitive issue when it comes to security.
While I'm always on the crusade for small APIs, can you pull Christian Haas into the loop?
He knows this stuff a lot better.